cvat/apps/engine/static/engine/js/shapeBuffer.js

```diff
@@ -185,14 +185,39 @@ class ShapeBufferModel extends Listener  {
                     object.frame ++;
                     numOfFrames --;
 
-                    // Propagate only for frames with same size
+                    object.z_order = this._collection.zOrder(object.frame).max;
+
                     let imageSize = imageSizes[object.frame - startFrame] || imageSizes[0];
                     if ((imageSize.width != originalImageSize.width) || (imageSize.height != originalImageSize.height)) {
-                        continue;
+                        let oldPosition = Object.assign({}, object);
+                        if (this._shape.type === 'box') {
+                            object.xtl = Math.clamp(object.xtl, 0, imageSize.width);
```

Comment:
Let's scale the shape in accordance with resolution of a target image.

Reply:
Done.